src/accelerate/accelerator.py

```diff
@@ -2116,7 +2144,9 @@ def prepare_optimizer(self, optimizer: torch.optim.Optimizer, device_placement=N
             return optimizer
         if device_placement is None:
             device_placement = self.device_placement
-        optimizer = AcceleratedOptimizer(optimizer, device_placement=device_placement, scaler=self.scaler)
+        # NOTE: Special case with MS-AMP we do *not* pass in the scaler, optimizer handles it for us
```

Comment:
Above, it was stated that

> MS-AMP requires `GradScaler` even with bf16 autocast w/ single GPU or DDP:

Probably not a contradiction, but I found it confusing.

Reply:
Made it a bit more clear